server/utils/interventionsFormError.ts

```diff
@@ -0,0 +1,64 @@
+import { InterventionsServiceError, InterventionsServiceValidationError } from '../services/interventionsService'
+import interventionsServiceErrorMessages from './interventionsServiceErrorMessages'
+import { FormValidationError } from './formValidationError'
+
+type InterventionsFormError = { type: 'general'; message: string } | ({ type: 'validation' } & FormValidationError)
```

Comment:
Let me know if this doesn't make sense, but I wonder if it's worth splitting this into two separate types of errors (validation and general) rather than adding a "type". The `validationErrorOrReturn` function introduced in the next commit feels a little bit to generalised (and a bit of a smell that it's doing too much).

Reply:
Would you be able to expand a little on what you think the code might look like? The `type` exists so that, ultimately, in `ReferralsController#convertError` we're able to determine whether we have a validation error (in which case we return it) or some other kind of error (in which case we throw it). Do you think there's an alternative way of structuring things that means we can lose the `type`?

Reply:
Having spent a bit too long thinking about this, the only alternative I've come up with would be to call this something like `interventionsFormValidationError` that basically does the `if` statement here, then in the call site (`validationErrorOrThrow`) you'd have:

```
if (interventionsServiceError.validationErrors) {
  return interventionsFormValidationError(interventionsServiceError)
} else {
  throw interventionsServiceError.message
}
```

I don't know if that's any better, it's basically doing the same thing but feels like _slightly_ less type checking? Feel free to ignore this though if you think it's not really an improvement.

Reply:
Yeah, I think I like this - it still gives us what we need and takes the re-throw logic out of the controller. I've pushed a change, is that what you had in mind?

Reply:
Yeah I think I prefer this, nicely done and thanks!